cmd/team-member-updater/main.go

```diff
@@ -0,0 +1,230 @@
+package main
+
+import (
+	"encoding/csv"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config/org"
+	"sigs.k8s.io/yaml"
+)
+
+type options struct {
+	spreadsheetPath string
+	configPath      string
+	logLevel        string
+}
+
+func (o *options) Validate() error {
+	if o.spreadsheetPath == "" {
+		return errors.New("required flag --spreadsheet was unset")
+	}
+
+	if o.configPath == "" {
+		return errors.New("required flag --config was unset")
+	}
+
+	level, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %v", err)
+	}
+	logrus.SetLevel(level)
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.spreadsheetPath, "spreadsheet", "", "Path to AOS Team Member Tracking spreadsheet.")
+	fs.StringVar(&o.configPath, "config", "", "Path to peribolos config to update.")
+	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+	return o
+}
+
+func main() {
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logrus.Fatalf("Invalid options: %v", err)
+	}
+
+	var orgConfig struct {
+		Orgs map[string]org.Config `json:"orgs,omitempty"`
+	}
+	rawConfig, err := ioutil.ReadFile(o.configPath)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to load peribolos org config.")
+	}
+
+	if err := yaml.Unmarshal(rawConfig, &orgConfig); err != nil {
+		logrus.WithError(err).Fatal("Failed to unmarshal peribolos org config.")
+	}
+
+	rawSpreadsheet, err := os.Open(o.spreadsheetPath)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to load org spreadsheet.")
+	}
+	defer func() {
+		if err := rawSpreadsheet.Close(); err != nil {
+			logrus.WithError(err).Fatal("Failed to close spreadsheet.")
+		}
+	}()
+
+	spreadsheetRecords, err := csv.NewReader(rawSpreadsheet).ReadAll()
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to parse spreadsheet.")
+	}
+
+	var members []teamMember
+	// ignore the header
+	for _, row := range spreadsheetRecords[1:] {
+		member := teamMemberFromRow(row)
+		if member.name == "" {
+			continue
+		}
+		if member.githubID == "" {
+			logrus.Warnf("No GitHub ID found for %s", member.name)
+			continue
+		}
+		members = append(members, member)
+	}
+
+	githubIDsbyGroup := map[string]sets.String{}
+	githubIDsbyTeam := map[string]sets.String{}
+	githubIDsbyRole := map[string]sets.String{}
+	allGithubIDs := sets.NewString(orgConfig.Orgs["openshift"].Members...)
+	for _, member := range members {
+		allGithubIDs.Insert(member.githubID)
+		if member.group != "" {
+			if githubIDsbyGroup[member.group] == nil {
```

Comment:
IMHO since this check is repeating also in other parts of the code, do you think that its better to have
```go
type githubIDs struct {
    byGroup map[string]sets.String
    byTeam  map[string]sets.String
    byRole  map[string]sets.String
}
```
and construct them?

or spliting the logic with a helper function?
```go
func populateGithubIDs(key,value string, set map[string]sets.String ) {
    if set[key] == nil {
        set[key] = sets.NewString(value)
    } else {
        set[key].Insert(value)
    }
}
```